analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsNull.cs

```diff
@@ -25,13 +25,27 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors
 {
     internal static class IsNull
     {
-        public static ProgramState Process(SymbolicContext context, IIsNullOperationWrapper isNull) =>
-            context.State[isNull.Operand] is { } value
-            && value.HasConstraint<ObjectConstraint>()
-                ? context.SetOperationConstraint(BoolConstraint.From(value.HasConstraint(ObjectConstraint.Null)))
-                : context.State;
+        public static ProgramState[] Process(SymbolicContext context, IIsNullOperationWrapper isNull)
```

Comment:
This whole logic now repeats 3x. I will refactor it in a separate PR after this and `IsType` are done